### ISSUE TITLE
`git lfs track` writes single quotes to .gitattributes on Windows

### ISSUE THREAD

**Issue by larsxschneider** [git-lfs/git-lfs#2258]
If you pass a tracking pattern with single quotes on Windows in `cmd.exe` then these single quotes are written to `.gitattributes`. This does not happen on macOS/Linux.

### Steps to reproduce
1. Open `cmd.exe`, navigate to a Git repo
2. Run:
```
git lfs track '*.bin'`
```
3. `.gitattributes` will look like this:
```
'*.bin' filter=lfs diff=lfs merge=lfs -text
```
I would expect this:
```
*.bin filter=lfs diff=lfs merge=lfs -text
```

### Environment
We can reproduce this issue on Win 7 and 8 (I haven't tested other versions) with Git LFS 2.1.1.

**Comment by technoweenie** [comment]
You're hitting a difference between bash and windows shell, not a platform inconsistency with LFS. Linux shells tend to expand path patterns to the current matches, so that the individual files are added and not the wildcard. Quotes are needed in linux shells to escape that pattern

```sh
# zsh, on my mac
$ git lfs track *.md
Tracking "README.md"
```

That said, trimming surrounding quotes isn't a bad idea. What about legit quotes in filenames though?

**Comment by larsxschneider** [comment]
Well, the same is true for Mac. I usually track patterns like this:
```
# also zsh on mac
$ git lfs track '*.bin'
Tracking "*.bin"
```
... and GitLFS creates the correct pattern without quotes in `.gitattributes`.

---

PS: I have no sympathy for people that put quotes into their filenames 🙈 😄

**Comment by technoweenie** [comment]
Right, that's because the shell is sending the exact string `*.bin` to Git LFS. Git LFS isn't stripping quotes on linux/mac, and then just forgetting to do the same on Windows.

It's just semantics though. From the user perspective, they see docs suggest running `git lfs track "*.foo"`, and type that. It should just work without requiring devs to the particulars of various shells.

I think as a compromise, we could only strip matching quotes at the beginning and end of the string. And if your file is literally `"track-this-lfs-lol.zip"`, then you can edit `.gitattributes` manually.

**Comment by vtbassmatt** [comment]
Another option is to switch all the examples/docs to use double quotes. I'm not in front of my Mac right now, but I think bash strips both double and single quotes. cmd.exe strips double quotes, so this would be relatively cross-platform.

**Comment by dashesy** [comment]
I had to use
    
    git lfs track "*.bin"

instead of 

    git lfs track *.bin

In git-bash on Windows, for lfs to track the files, in the meanwhile `git lfs status` showed the tracked files before commit but `git lfs ls-files` was empty after commit. When I added the quotes, it worked.
